cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive12ExploreService.java

```diff
@@ -129,4 +135,29 @@ protected OperationHandle doExecute(SessionHandle sessionHandle, String statemen
     throws HiveSQLException, ExploreException {
     return getCliService().executeStatementAsync(sessionHandle, statement, ImmutableMap.<String, String>of());
   }
+
+  @Override
+  protected SessionHandle doOpenHiveSession(Map<String, String> sessionConf) throws HiveSQLException {
+    SessionHandle sessionHandle = super.doOpenHiveSession(sessionConf);
+
+    // HDP Hive 12 only sets SessionState when an operation is executed.
+    // The SessionState is not set during an openSession call.
+    // Hence execute a dummy statement during openSession, so that SessionState gets set.
+    OperationHandle opHandle = null;
+    try {
+      opHandle = getCliService().executeStatement(sessionHandle, "set cdap_tmp_var=1",
+                                                  Collections.<String, String>emptyMap());
+    } catch (Throwable t) {
+      LOG.error("Error setting temporary variable to set session state", t);
```

Comment:
Yes, it is okay to ignore the exception since the session is started as soon as the `executeStatement` is called. So, there is a high chance things will work. If for some reason session is not started, then later we'll throw exception anyways, and query will fail.